### PR TITLE
Added `setPublicKey` method to set public key to the user in case the public key is null.

### PR DESCRIPTION
**Pull request by Mahmoud-Emad**
### Related issues
- https://github.com/threefoldtech/rmb-sdk-ts/issues/40

### CODE REVIEW

File: lib/client.ts [hunk 1]
```diff
@@ -93,15 +93,21 @@ class Client {
...
+                        this.relayUrl,
+                        this.keypairType
+                    ],
+                )
             }
-
             this.updateSource();
             this.createConnection()
-
+
```
Comment (AhmedHanafy725) [inline lib/client.ts]: please remove this white space

File: lib/client.ts [hunk 2]
```diff
@@ -93,15 +93,21 @@ class Client {
         try {
             await this._initApi();
             await this.createSigner();
-            this.twin = await getTwinFromTwinAddress(this.api!, this.signer.address)
-
+            this.twin = await getTwinFromTwinAddress(this.api!, this.signer.address)            
             if (!this.twin.pk) {
-                this.twin.pk = getPublicKey(this.mnemonics);
+                await applyExtrinsic(
```
Comment (AhmedHanafy725) [inline lib/client.ts:99]: I think after setting the pk on the twin on the chain, we have to update `this.twin` to include the pk on the twin object

Comment (mahmoud-emad) [reply lib/client.ts:99]: Updated

File: lib/client.ts (hunk 1 above)
Comment (mahmoud-emad) [reply lib/client.ts]: Updated

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**